### PR TITLE
Windows usage improvements

### PR DESCRIPTION
**Pull request by mantazzo**
General improvements to the code to make it working better on Windows systems, while keeping it available for Linux. Most changes should be cross-platform. Haven't fixed everything, but it's working more or less. Still requires to run "php esc run" twice (first to set up, 2nd to actually run) to make sure it works.
Haven't tested other things like commands (server on LAN, account is only Starter Access), but it should be fine I think.

Signed-off-by: mantazzo <mantas.jokubauskas@gmail.com>

### CODE REVIEW

File: core/Classes/File.php [hunk 1]
```diff
@@ -224,7 +224,7 @@ public static function makeDir(string $dir)
     {
         $dir = str_replace('/', DIRECTORY_SEPARATOR, $dir);
 
-        if ((isWindows() && !preg_match('/^\w:(\\|\/)/', $dir)) || (!isWindows() && substr($dir, 0, 1) != DIRECTORY_SEPARATOR)) {
```
Comment (brakerb) [inline core/Classes/File.php:227]: Why did you remove the `!`? Now the statement is true when the path starts with C:/ or D:\ I see that I forgott an i to make it case-insensitive

Comment (mantazzo) [reply core/Classes/File.php:227]: Ah, I forgot to include a comment there - for some reason, in a new install, it fails with the ! sign added (in MapController.php, line 53 - I see I included a comment there). Without the sign, it worked fine. I am honestly not sure why it didn't work, but maybe it's just for me.

**Comment by mantazzo** [comment]
Pushed a (potential) fix to the preg_match.

File: core/Classes/File.php (hunk 1 above)
Comment (brakerb) [reply core/Classes/File.php:227]: That is because the regex is currently case-sensitive and only checks for c, d not C, D. I need to make it //i

**Comment by brakerb** [comment]
Oh and in php a regex needs to start and end with the same character. Using ## can be helpful when using slashes in your regex. You could also do `/^[a-z]:(\\|\/)/i` it would have the same effect.
